src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -108,7 +108,7 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
               if (method ne null)
                 return method
               else {
-                method = ScalaRunTime.ensureAccessible(forReceiver.getMethod("xyz", methodCache.parameterTypes()))
+                method = ScalaRunTime.lookupStructuralMethod(forReceiver, "xyz", methodCache.parameterTypes())
```

Comment:
We could also emit the entire body of `lookupStructuralMethod` in CleanUp in 2.12.x, and introduce the ScalaRunTime method only in 2.13.x